tests/test_webdriver_remote.py

```diff
@@ -51,9 +52,169 @@ def setUp(self):
 
     def test_support_with_statement(self):
         "Remote should support with statement"
-        with Browser("remote"):
+        with Browser("remote", browser="chrome"):
             pass
 
     def test_should_be_able_to_change_user_agent(self):
         "Remote should not support custom user agent"
         pass
+
+
+@pytest.mark.macos
+class RemoteBrowserSafariTest(WebDriverTests, unittest.TestCase):
+    @pytest.fixture(autouse=True, scope="class")
+    def setup_browser(self, request):
+        # test with statement. It can't be used as simple test
+        # because safari doesn't support multisessions
+        with Browser("remote", browser="safari"):
+            pass
+
+        request.cls.browser = Browser("remote", browser="safari")
+        request.addfinalizer(request.cls.browser.quit)
+
+    def setUp(self):
+        self.browser.visit(EXAMPLE_APP)
+
+    def test_support_with_statement(self):
+        """
+        Remote should support with statement
+        See setup browser
+        """
+
+    def test_should_be_able_to_change_user_agent(self):
+        "Remote should not support custom user agent"
+        pass
```

Comment:
can we explicitly skip it with a message?

Reply:
Added